### PR TITLE
Re-set the enketo form's language when a DOM node is added to the form

### PR DESCRIPTION
**Pull request by m5r**
# Description

Leverage [MutationObserver](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver#browser_compatibility) to react to DOM changes to the form and re-set the enketo form's language.  
~~Using MutationObserver resulted in infinite loops in some cases where it rendered the app completely unresponsive, which is why I resorted to listening to form changes. This is not ideal but it works™ and forms are still responsive when editing them with an HTC One S.~~ Edit: see [comment](https://github.com/medic/cht-core/pull/7520#issuecomment-1054364074)  
This fix won't be needed anymore once the enketo uplift gets merged medic/cht-core#7256

medic/cht-core#7515

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Comment by m5r** [comment]
@jkuester can you please review this fix please? Thanks!

File: tests/e2e/forms/repeat-form.wdio-spec.js [hunk 1]
```diff
@@ -7,16 +7,29 @@ const loginPage = require('../../page-objects/login/login.wdio.page');
...
+  internalId: 'repeat-translation-button',
+  title: 'Repeat button',
+  type: 'form',
+  _attachments: {
+    xml: {
+      content_type: 'application/octet-stream',
+      data: Buffer.from(buttonForm).toString('base64')
     }
   }
 };
```
Comment (garethbowen) [inline tests/e2e/forms/repeat-form.wdio-spec.js:35]: This is mostly a repeat of the lines above - let's make a function that takes the filename, id, and title (or make the title the same as the id?) to save the duplication.

File: webapp/src/ts/services/enketo.service.ts [hunk 2]
```diff
@@ -292,12 +292,18 @@ export class EnketoService {
           return Promise.reject(new Error(JSON.stringify(loadErrors)));
         }
         const language = options.language;
+        this.currentForm.langs.$formLanguages.val(language).trigger('change');
+        // re-set the enketo form's language when a DOM node is added/removed to the form
```
Comment (garethbowen) [inline webapp/src/ts/services/enketo.service.ts:296]: I think this comment is out of date. If I'm understanding correctly, the `change` event fires when an input in the form is updated by the user, not when a DOM node is added or removed.

File: webapp/tests/karma/ts/services/enketo.service.spec.ts [hunk 3]
```diff
@@ -95,6 +95,7 @@ describe('Enketo service', () => {
     EnketoForm.returns({
       view: {
         $: { on: sinon.stub() },
+        html: document.createElement('div'),
```
Comment (garethbowen) [inline webapp/tests/karma/ts/services/enketo.service.spec.ts:98]: I don't think you need this any more.

**Review by garethbowen — CHANGES_REQUESTED** [review]
As Josh is on leave and this is a pressing issue, I've jumped in to review - hope that's ok!

This looks great. The change event will fire quite often - have you checked that the performance is ok?

Other than that I have a few comments inline but nothing major.

**Review by jkuester — COMMENTED** [review]
I did have a look through these code changes since they are dealing with the Enketo functionality. I don't really have anything to add besides what Gareth has already pointed out, but I did test your new `repeat-translation-count` form with the new Enketo version and verified that it still works as expected.  :+1:

**Comment by m5r** [comment]
Thanks for the feedback @garethbowen @jkuester

I was frustrated with the solution I came up with and thought about it all weekend :joy:  
I decided to re-introduce the `MutationObserver` to trigger the language change when both conditions are met:
- a DOM node is added to the form -- `MutationObserver` picked up at least one new node in the form
- the form has been edited -- the form `change` event was triggered
This fixes our translation issue but also put less strain on the phones' CPU by asking Enketo fewer times to change the form's language

**Review by garethbowen — APPROVED** [review]
The new approach looks reasonable. I worry with state like this that further changes may regress if someone forgets to set it, but because we'll remove the code in 4.0.0 I think the risk is minimal.

**Review by jkuester — APPROVED** [review]

**Pull request merged**